File: climate/paleoclimate/Petit01.jl
using CairoMakie, Makie.Colors
using CSV, DataFrames

CairoMakie.activate!(; px_per_unit = 2)

df = DataFrame(CSV.File("./DeuteriumVostok.csv"; delim=','))
time = df[!,1]
SST = df[!,2]

fig = Figure(; size = (1200, 800), backgroundcolor = :transparent)
ax = Axis(fig[1,1:4],
    # Background
    backgroundcolor = :transparent,
    xgridvisible = false,
    ygridvisible = false,
    #limits = ((0,10),(18,27)),
    # Title
    title = "end of the last glaciation",
    titlevisible = true,
    titlesize = 50,
    titlealign = :center,
    titlegap = 4.0,
    # x-axis
    xlabel = "kyears BCE",
    xlabelvisible = true,
    xlabelsize = 50,
    xlabelcolor = :black,
    xlabelpadding = -60.0,
    xticks = [60,50,20,10],
    xticksvisible = true,
    xticksize = 20,
    xticklabelsvisible = true,
    xticklabelsize = 50,
    xtickformat = "{:.0f}",
    xscale = identity, #log10,
    xaxisposition = :bottom,
    # y-axis
    ylabel = "Deuterium",
    ylabelvisible = true,
    ylabelsize = 50,
    ylabelcolor = :black,
    ylabelpadding = -100.0,
    yticks = [-490,-480,-435,-425],
    yticksvisible = true,
    yticksize = 20,
    yticklabelsvisible = true,
    yticklabelsize = 50,
    #ytickformat = "{:.1f}",
    yscale = identity,
    yaxisposition = :left,
)
ax.xreversed = true
#text!(13, -480, text = L"\text{(b)}", align = [:left, :top], color = :black, fontsize = 70)
#text!(9, 1, text = L"\times 10 ^{-3}", align = [:right, :bottom], color = :black, fontsize = 30)
lines!(ax, time.*1e-3, SST, color = :red, linewidth = 3)
save("../../results/climate/paleoclimate/Petit01.png", fig)

File: climate/paleoclimate/Petit01.py
from pandas import read_csv
from matplotlib import pyplot as plt

# Render LaTeX to Matplotlib (https://matplotlib.org/stable/gallery/text_labels_and_annotations/tex_demo.html)
# needs extra packages (https://stackoverflow.com/questions/11354149/python-unable-to-render-tex-in-matplotlib)
plt.rcParams.update({
    "text.usetex": True,
    "font.family": "sans-serif",
    })

timeseries = read_csv('DeuteriumVostok.csv', sep = ',', engine = 'python', header=0, index_col=0)
print(type(timeseries))
print(timeseries.head)

fig = plt.figure(figsize=[12.8,9.6], dpi=200, layout='tight')
plt.gca().invert_xaxis()
plt.plot(timeseries, linewidth = 4, color='red')
plt.xlabel(r'yr', fontsize = 40)
plt.ylabel(r'$2$H concentration', fontsize = 40)
plt.xticks(fontsize = 35, weight = 'bold')
plt.yticks(fontsize = 35, weight = 'bold')
plt.title('end of the last glaciation', fontsize = 40)
plt.savefig("../../results/climate/paleoclimate/Petit01.png", transparent=True, dpi=600)
